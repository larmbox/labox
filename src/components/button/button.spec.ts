import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LButton } from '~/index';

test('mount component', async () => {
  expect(LButton).toBeTruthy();

  const wrapper = mount(LButton, {
    props: {
      id: 'id',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.html()).toContain('button button-md button-primary');
});

test('register css variables', async () => {
  mount(LButton);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-button');
});

test('variant', async () => {
  const component = mount(LButton, {
    props: {
      id: 'id',
      variant: 'secondary',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('button-secondary');

  await component.setProps({ variant: 'error' });
  expect(component.html()).toContain('button-error');
});

test('disabled', async () => {
  const component = mount(LButton, { props: { id: 'button' } });

  expect(component.html()).not.toContain('disabled');

  await component.setProps({ disabled: true });
  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('disabled');
});

test('link types', async () => {
  const component = mount(LButton, {
    props: { id: 'button', href: '/', tag: 'a' },
  });
  expect(component.html()).toMatchSnapshot();
});

test('loading', async () => {
  const component = mount(LButton, {
    props: { loading: true },
  });
  expect(component.html()).toContain('loading-inherit');
});

test('inherit attributes', async () => {
  const component = mount(LButton, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
