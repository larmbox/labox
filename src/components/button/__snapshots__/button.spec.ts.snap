// Vitest Snapshot v1

exports[`disabled 1`] = `
"<button id=\\"button\\" class=\\"button button-md button-primary\\" disabled=\\"true\\">
  <!--v-if-->
  <div class=\\"button-content\\">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`link types 1`] = `
"<a id=\\"button\\" class=\\"button button-md button-primary\\" href=\\"/\\">
  <!--v-if-->
  <div class=\\"button-content\\">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</a>"
`;

exports[`mount component 1`] = `
"<button id=\\"id\\" class=\\"button button-md button-primary\\">
  <!--v-if-->
  <div class=\\"button-content\\">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;

exports[`variant 1`] = `
"<button id=\\"id\\" class=\\"button button-md button-secondary\\">
  <!--v-if-->
  <div class=\\"button-content\\">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</button>"
`;
